relentless/potential/potential.py

```diff
@@ -391,14 +416,27 @@ def derivative(self, pair, param, r):
         r,deriv,scalar_r = self._zeros(r)
         if any(r < 0):
             raise ValueError('r cannot be negative')
+        if not isinstance(var, Variable):
+            raise TypeError('Parameter with respect to which to take the derivative must be a Variable.')
 
         # only evaluate at points inside [rmin,rmax], if specified
         flags = np.ones(r.shape[0], dtype=bool)
         if params['rmin'] is not False:
             flags[r < params['rmin']] = False
         if params['rmax'] is not False:
             flags[r > params['rmax']] = False
-        deriv[flags] = self._derivative(param, r[flags], **params)
+
+        # evaluate with respect to parameter
+        deriv[flags] = 0
+        d_params = [i for i in self.coeff.params if not(i=='rmin' or i=='rmax' or i=='shift')]
+        for p in d_params:
+            p_obj = self.coeff[pair][p]
+            if isinstance(p_obj, DependentVariable):
+                deriv[flags] += (p_obj.derivative(var)
+                                *self._derivative(p, r[flags], **params))
```

Comment:
To make the intent clear (as a chain rule), can you reverse the order of these operations?